### PR TITLE
add notes on MD dictionary

### LINKED ISSUE #36: centralize notes on MD
Centralize notes on the MD database, pulling information from this issue:

https://github.com/UAlbertaALTLab/dictionary-database/issues/5

and placing it in this document:

https://github.com/UAlbertaALTLab/dictionary-database/blob/main/docs/MD.md

### PR DESCRIPTION
**Pull request by dwhieb**
This PR compiles notes from various places into a single document about the Maskwacîs dictionary.

closes #36

### CODE REVIEW

File: docs/CW.md [hunk 1]
```diff
@@ -146,17 +144,13 @@ A list of forms showing the derivational breakdown of the word, each separated b
 
 The date that the entry was last updated.
 
-## `\fststem` **FST stem**
-
-This field is _not_ in the Toolbox database. This data only lives in the TSV version of this file. There are somewhere in the range of ~1,500 entries total where this information has been added. This data should be retrieved and added to the main entry in the dictionary database. Forms marked as "CHECK" are ones that need FST stems.
```
Comment (aarppe) [inline docs/CW.md:151]: For FST stems marked with CHECK, we hadn't been able to resolve the FST stem. A typical reason is that the entries in question are derived or inflected forms of a lemma, and perhaps should not be considered as a regular lemma.

File: docs/CW.md [hunk 2]
```diff
@@ -197,15 +189,15 @@ Arok is in the process of adding this field to the database. [2021/03/16] This f
...
+The value of this field is always `new` (with perhaps one exception; see below). This field indicates that the entry was added after the publication of the 2001 edition of the _Cree: Words_ dictionary was published.
 
-In at least one case, this field appears multiple times, with one of the fields blank, and the other containing `new`.
+Not all entries since 2001 are marked as new. These could potentially be determined based on information in the `\src` field. In at least one case, this field appears multiple times, with one of the fields blank, and the other containing `new`.
 
 ## `\ps` **part of speech** [multiple]
 
-This field really combines information on part of speech, morpheme type, and inflectional class.
+This field really combines information on part of speech, morpheme type, and inflectional class. If an entry has multiple `\ps` fields, those should each be given their own sense (with the same definition, if necessary).
```
Comment (aarppe) [inline docs/CW.md:200]: Note that for Indeclinable Particles, the "inflectional class" is rather a subtype of a particle than a real inflectional class.

File: docs/MD.md [hunk 3]
```diff
@@ -1,29 +1,129 @@
...
-The TSV version of the Maskwacîs dictionary in the ALTLab repo also has semantic classifications, using the Rapid Words ontology.
+Notes on the Maskwacîs dictionary database.
 
 ## SRO
 
-* Inconsistent marking of vowel length.
+A transcription of the headword in Standard Roman Orthography (SRO).
+
+* Occasionally includes punctuation, **ex:** _ah?_ 'What? What did you say?'.
+* Vowel length is indicated with double vowels rather than macrons or circumflexes, **ex:** `aa`.
```
Comment (aarppe) [inline docs/MD.md:10]: Nope, as far as I know. Rather, long vowels lack diacritical marking.

File: docs/MD.md [hunk 4]
```diff
@@ -1,29 +1,129 @@
...
+* Vowel length is not consistently indicated. Many long vowels are written as short vowels.
```
Comment (aarppe) [inline docs/MD.md:11]: Vowel length is not indicated at all in MD, as we originally received the spreadsheet.

File: docs/MD.md [hunk 5]
```diff
@@ -1,29 +1,129 @@
...
+* `<i>` is (sometimes?/always?) used for unstressed short vowels (or maybe just unstressed short /a/?).
```
Comment (aarppe) [inline docs/MD.md:12]: The right adverb is 'sometimes' for <i> being used for the unstressed vowel.

File: docs/MD.md [hunk 6]
```diff
@@ -1,29 +1,129 @@
...
+  - **ex:** MD _wapimew_ = SRO _wâpamêw_
+* /h/ is sometimes omitted before stops (/c, p, t, k/).
+* /ou/ is written `<iw>` or `<ow>`.
+* SRO `<c>` is sometimes written as `<ch>` or `<ts>`.
+* SRO `<c>` and `<ci>` can both be written as `<ts>`.
+  - **ex:** MD _mitsow_ = SRO _mîcisow_
+* Very occasionally includes a hyphen.
+* Some entries are multiword headwords, **ex:** _ayamihew masinahikan_.
+* Imperatives are not always given in their lemma form. This might need to be reconstructed using the FST.
```
Comment (aarppe) [inline docs/MD.md:21]: What you mean is that there isn't always also a lemma entry for imperative verb entries.

In some case we may need to reconstruct the lemma ourselves.

File: docs/MD.md [hunk 7]
```diff
@@ -1,29 +1,129 @@
...
+reply                            | reply
+suffix                           | suffix
+v                                | verb
+v comb                           |
+v command                        | imperative verb
+v phrase                         |
+v phrase pl                      |
+v pl                             | plural-only verb
+v sg                             | singular-only verb
+vc                               |
```
Comment (aarppe) [inline docs/MD.md:74]: command verb, i.e. imperative

File: docs/MD.md [hunk 8]
```diff
@@ -1,29 +1,129 @@
...
+
+## MeaningInEnglish
 
-## Definition
+The English definition of the term.
 
+* Occasionally includes example phrases or sentences, usually preceded by `e.g.` / `E.g.`.
+  - Other types of notes are also preceded by `e.g.` / `E.g.`, making it difficult to rely on this for programmatic processing.
+  - Example phrases are followed by their translations. If the example ends in punctuation, the translation simply follows the example, after a space. If the example does not end in punctuation, the example and its translation are separated by a hyphen surrounded by spaces.
+  - **ex:** "You are cold. Usually used in a question. E.g. kikawacin ci? Are you cold?"
```
Comment (aarppe) [inline docs/MD.md:90]: The orthography of these example words/sentences are not according to SRO (in particular vowel length).

File: docs/MD.md [hunk 9]
```diff
@@ -1,29 +1,129 @@
...
   - _original_: the original MD definition, verbatim
     - Store this in the alternative analysis entry.
   - _FST_: the cleaned and edited version sent to the FST
     - Store this in the main entry, on the sense from MD.
   - _user_: original, except with Cree words corrected
     - Store this in the main entry.
+
+## RapidWordsClasses
+
+A semicolon-delimited list of rapid word classes that apply to this entry. Each class corresponds to a index in the [RapidWordsIndices](#RapidWordsIndices) field.
```
Comment (aarppe) [inline docs/MD.md:113]: The RW classes are given at the second-highest and deepest possible level of the RW hierarchy (to the extent possible). Entries may have been judged to fall under more than one RW class (at the second-highest and deepest level in the RW hierarchy).

File: docs/MD.md [hunk 10]
```diff
@@ -1,29 +1,129 @@
...
+
+## RapidWordsIndices
+
+A semicolon-delimited list of indices for the rapid word classes that apply to this entry. Each index corresponds to a rapid words class in the [RapidWordsClasses](#RapidWordsClasses) field.
```
Comment (aarppe) [inline docs/MD.md:117]: The same applies here: the indices are given at the second highest and deepest possible levels in the RW hierarchy, and an entry can belong under multiple distinct classes.

File: docs/MD.md [hunk 11]
```diff
@@ -1,29 +1,129 @@
...
+
+## English_POS
+
+The English definition, with each word tagged for part of speech.
```
Comment (aarppe) [inline docs/MD.md:121]: The parsing was undertaken with the publicly available Stanford parser. That provides a lemma for word-forms, which is provided for verbs (and nouns, I think).

File: docs/MD.md [hunk 12]
```diff
@@ -1,29 +1,129 @@
...
+
+* The tag is appended to the end of each word in the format `_{TAG}`, **ex:** `The_DET`, indicating that the word _The_ is a determiner.
+* Items with no tag have the appended underscore but no tag.
+* Punctuation is also tagged with just the underscore, and no tag.
+
+## English_Search
+
+Contains a simplified version of the definition. The use of this field and how it was generated is currently unclear.
```
Comment (aarppe) [inline docs/MD.md:129]: For verbs, this means providing an English-like entry starting with the infinitive form of the verb (using the lemma, and excluding the initial subject pronoun).

**Review by aarppe — COMMENTED** [review]
Went over and provided comments.

File: docs/MD.md (hunk 3 above)
Comment (dwhieb) [reply docs/MD.md:10]: I found four instances of doubled vowels in the data:

* `ikatestaa	ᐃᑲᑌᐢᑕᐊ	vc	Put it aside somewhere else.`
* `sihtoskaawew	ᓯᐦᑐᐢᑲᐊᐁᐧᐤ	vp	He supports him.`
* `kiipahikan	ᑭᐃᐸᐦᐃᑲᐣ	n	A lid, cork or stopper.`
* `kiipiskah	ᑭᐃᐱᐢᑲᐦ	vc	Block the way.`
* `kiisapiskisam	ᑭᐃᓴᐱᐢᑭᓴᒼ	v	He heats a metal object.`

File: docs/MD.md (hunk 4 above)
Comment (dwhieb) [reply docs/MD.md:11]: I found four instances of doubled vowels in the data:

* `ikatestaa	ᐃᑲᑌᐢᑕᐊ	vc	Put it aside somewhere else.`
* `sihtoskaawew	ᓯᐦᑐᐢᑲᐊᐁᐧᐤ	vp	He supports him.`
* `kiipahikan	ᑭᐃᐸᐦᐃᑲᐣ	n	A lid, cork or stopper.`
* `kiipiskah	ᑭᐃᐱᐢᑲᐦ	vc	Block the way.`
* `kiisapiskisam	ᑭᐃᓴᐱᐢᑭᓴᒼ	v	He heats a metal object.`

It seems safe to assume those are intended as long vowels? Although I notice that the Syllabic transcriptions of all of these items do _not_ have long vowels.

File: docs/MD.md (hunk 8 above)
Comment (dwhieb) [reply docs/MD.md:90]: Right—we'll need to identify stretches of Cree within definitions and standardize them.

File: docs/MD.md (hunk 4 above)
Comment (aarppe) [reply docs/MD.md:11]: Yes, one could presume that's the intention, though one could also consider them typos.

I was able to match all of this with CW content in itwêwina, and they're not always spelled with the long vowel, e.g.:

https://itwewina.altlab.app/search?q=ikatestaa
https://itwewina.altlab.app/search?q=kisapiskisam

To really determine the vowel length, we'd need to listen to the recording as well as match with the linguistic analysis.

**Pull request merged**